.github/workflows/publish.yml

```diff
@@ -0,0 +1,68 @@
+name: 🚀 Publish to the Visual Studio Marketplace
+on:
+  push:
+    branches:
+      - main
+
+jobs:
+  publish:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check it
+        uses: actions/checkout@v4
+
+      - name: Prepare it
+        uses: actions/setup-node@v4
+        with:
+          node-version: lts/*
+
+      - name: Install it
+        run: npm clean-install
+
+      - name: Package it
+        id: package
+        run: |
+          npm exec vsce package
+          echo ::set-output filename=vsix::$(ls *.vsix)
+
+      - name: Publish it
+        id: publish
+        env:
+          VSCE_PAT: ${{ secrets.VSCE_PAT }}
+        run: npm exec vsce publish -- --packagePath ${{ steps.package.outputs.filename }}
+
+      - name: Abort it
+        if: ${{ steps.publish.outcome == 'failure' }}
+        run: |
+          echo "Publish failed. Release may already exist."
+          exit 1
```

Comment:
No need to abort, if the action will implicitly check if the previous step failed anyways.
I wanted to not explicitly fail in the event the version hasn't been updated, but can't think of a good way to implement this. Will remove it for now